### PR TITLE
Fix related products E2Es for WC9.5

### PR DESCRIPTION
**Pull request by tomalec**
### Changes proposed in this Pull Request:

This PR adjusts E2E tests for WC 9.5 changes.

In WC 9.5, by default, there is no related products section on a single product page - I made the test skip in such a case.
Also, I added a "related" tag to all simple products to make sure simple products will appear in the related products section instead of "Cat teasers" - without the "Add to cart" button.

### Screenshots:

image

### Detailed test instructions:

1. Run E2E tests locally
2. Run [E2E workflow](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/workflows/e2e-tests.yml) with `9.5.0-rc.1` - https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12362867871
3. Run [E2E workflow](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/workflows/e2e-tests.yml) with `9.4.0` - https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12383008805
4. Review the code to check the tests are testing what they should

### Additional details:

- I'll make a separate PR with a proposal to simplify and DRY a bit our `create*Page`s

### Changelog entry

> Dev - Improve E2E tests for related products.

### CODE REVIEW

**Comment by eason9487** [comment]
> 3\. Run [E2E workflow](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/workflows/e2e-tests.yml) with `9.4.0` - https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12363338232

This run didn't pass. I got the same failure in a local run.

**Comment by tomalec** [comment]
Thanks. I forgot to migrate the adjusted selector. Now, the run is passing https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/12383008805

File: tests/e2e/utils/create-page.js [hunk 1]
```diff
@@ -135,3 +136,18 @@ export async function createAllProductsBlockShopPage() {
 		( await createPage( title, pageContent ) )
 	);
 }
+
+/**
+ * Creates a page with related products block.
+ *
+ * @param {number} productId Product ID.
+ * @return {string} Slug of the created page.
```
Comment (eason9487) [inline tests/e2e/utils/create-page.js]:
```suggestion
 * @return {Promise<string>} Slug of the created page.
```

**Review by eason9487 — APPROVED** [review]
Thanks for the compatibility fix! LGTM.

**Comment by eason9487** [comment]
Hi @tomalec, I will address https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/481#discussion_r1889560287 and merge this PR to make the later release process a bit smoother.

**Pull request merged**